pkg/main.go

```diff
@@ -12,18 +15,24 @@ import (
 
 func main() {
 	// Start listening to requests sent from Grafana.
-	s := athena.New()
-	ds := awsds.NewAsyncAWSDatasource(s)
-	ds.Completable = s
-	ds.EnableMultipleConnections = true
-	ds.CustomRoutes = routes.New(s).Routes()
-
 	if err := datasource.Manage(
 		"grafana-athena-datasource",
-		ds.NewDatasource,
+		MakeDatasourceFactory(),
 		datasource.ManageOpts{},
 	); err != nil {
 		log.DefaultLogger.Error(err.Error())
 		os.Exit(1)
 	}
 }
+
+func MakeDatasourceFactory() datasource.InstanceFactoryFunc {
+	s := athena.New()
```

Comment:
Out of curiosity, is there a reason that `s` is declared outside of the return func?

Reply:
Good question - I intended to keep part of the current behavior where the `AthenaDatasourceIface` is shared, but looking more at it I think that isn't particularly helpful, and it'll be safer not to share it. I'll move it in.